Tests/MongoSwiftSyncTests/SyncMongoClientTests.swift

```diff
@@ -213,4 +213,39 @@ final class SyncMongoClientTests: MongoSwiftTestCase {
 
         try defaultDb.drop()
     }
+
+    // Ensure that sync clients stay alive as long as their child objects are still in scope.
+    func testClientLifetimeManagement() throws {
```

Comment:
this is basically a regression test for what I stated above about keeping clients alive for the appropriate amount of time.